goal_src/engine/ui/credits.gc

```diff
@@ -5,3 +5,240 @@
 ;; name in dgo: credits
 ;; dgos: GAME, ENGINE
 
+;; definition for function set-credits-font-color
+;; INFO: Return type mismatch int vs none.
+(defun set-credits-font-color ((arg0 float))
+  (let ((f0-0 255.0))
+   )
+  (dotimes (v1-0 4)
+   (let ((f0-2 (* 64.0 arg0)))
+    (if (< 128.0 f0-2)
+     (set! f0-2 128.0)
+     )
+    (let
+     ((a1-6
+       (logior
+        (logand
+         (l.wu (+ (+ (* v1-0 4) (the-as int *font-work*)) 2496))
```

Comment:
this fails to compile, maybe needs a nudge with a type cast?  This is a global and im pretty sure it is indeed a `font-work` not a numeric type as the code here is trying to coerce it into

Reply:
I'm not sure what's going on here. This is setting a bitfield in a `char-color` in the `color-table` of the font work, but the decompiler can't figure out the load from the color-table.  I probably won't have a chance to take a closer look for a while, so I'm making an issue here: https://github.com/water111/jak-project/issues/538

Reply:
This should be fixed now.  Previously:
```lisp
(let
 ((a1-6
   (logior
    (logand
     (l.wu (+ (+ (* v1-0 4) (the-as int *font-work*)) 2496))
     (the-as uint -256)
     )
    (the-as uint (shr (shl (the int f0-2) 56) 56))
    )
   )
  (a2-6 (+ (* v1-0 4) (the-as int *font-work*)))
  )
 (s.w! (+ a2-6 2496) a1-6)
 )
```

now:
```lisp
(set! (-> *font-work* color-table 32 color v1-0 r) (the int f0-2))
```